### ISSUE TITLE
Make the CommandValue ordering passed to protocol driver be the same as the resource operation ordering in Device Profile

### ISSUE THREAD

**Issue by cloudxxx8** [edgexfoundry/device-sdk-go#360]
For some example, the protocol driver needs the specific ordering for some reason.  The following example should make sure "Action" is the first CommandValue element passed to protocol driver

```
deviceCommands:
  -
    name: AddUser
    set:
      - { object: "Action", parameter: "Add" }
      - { object: "UserName" }
      - { object: "Password" }
      - { object: "Privilege" }
   -
    name: DeleteUser
    set:
    - { object: "Action", parameter: "Delete" }
    - { object: "UserName" }
```

**Comment by cloudxxx8** [comment]
Another topic we could discuss here is whether we need "index" field.  I thought we can just follow the order and index looks redundant to me.  For example:

```
deviceCommands:
  -
    name: AddUser
    set:
      - { index: "1", object: "Action", parameter: "Add" }
      - { index: "2", object: "UserName" }
      - { index: "3", object: "Password" }
      - { index: "4", object: "Privilege" }
   -
    name: DeleteUser
    set:
    - { index: "1", object: "Action", parameter: "Delete" }
    - { index: "2", object: "UserName" }
```

**Comment by iain-anderson** [comment]
If the  ordering is preserved by core-metadata, agreed this should be fine and we don't need the index field.

**Comment by cloudxxx8** [comment]
fixed by #356
